python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -618,3 +656,121 @@ def measureMeanVarCov(self, exposure1, exposure2, region=None):
+            First exposure of flat field pair.
+
+        exposure2 : `lsst.afw.image.exposure.ExposureF`
+            Second exposure of flat field pair.
+
+        correctionType : `str`, optional
+            The correction applied, one of ['NONE', 'SIMPLE', 'FULL']
+
+        readNoise : `float`, optional
+            Amplifier readout noise (ADU).
+
+        region : `lsst.geom.Box2I`, optional
+            Region of each exposure where to perform the calculations
+            (e.g, an amplifier).
+
+        Returns
+        -------
+        gain : `float`
+            Gain, in e/ADU.
+
+        Raises
+        ------
+            RuntimeError: if `correctionType` is not one of 'NONE',
+                'SIMPLE', or 'FULL'.
+
+            RuntimeError: if a readout noise value is not provided
+                when `correctionType` is different from 'NONE'.
+        """
+        if correctionType not in ['NONE', 'SIMPLE', 'FULL']:
+            raise RuntimeError("Unknown correction type %s" % correctionType)
+
+        if correctionType != 'NONE' and readNoise is None:
+            raise RuntimeError("Must supply a readout noise value if "
+                               "performing gain correction.")
+
+        if region is not None:
+            im1Area = exposure1.getImage()[region].getArray()
+            im2Area = exposure2.getImage()[region].getArray()
+        else:
+            im1Area = exposure1.getImage().getArray()
+            im2Area = exposure2.getImage().getArray()
+
+        const = np.mean((im1Area - im2Area)**2 / (im1Area + im2Area))
+        gain = 1. / const
+
+        mu = 0.5*(np.mean(im1Area) + np.mean(im2Area))
+
+        if correctionType == 'SIMPLE':
+            gain = 1/(const - (1/mu)*(readNoise**2 - (1/2*gain**2)))
+        elif correctionType == 'FULL':
+            root = np.sqrt(mu**2 - 2*mu*const + 2*readNoise**2)
+            denom = (2*const*mu - 2*readNoise**2)
+
+            positiveSolution = (root + mu)/denom
+            negativeSolution = (mu - root)/denom  # noqa: F841 unused, but the other solution
+
+            gain = positiveSolution
+
+        return gain
+
+    def getReadNoiseFromMetadata(self, taskMetadata, ampName):
+        """Gets readout noise for an amp from ISR metadata.
+
+        Parameters
+        ----------
+        taskMetadata : `list`[`lsst.pipe.base.TaskMetadata`]
+                    List of exposures metadata from ISR.
+
+        ampName : `str`
+            Amplifier name.
+
+        Returns
+        -------
+        readNoise : `float`
+            Median of the overscan readnoise in the
+            post-ISR metadata of the input exposures (ADU).
+        """
+        # Empirical readout noise [ADU] measured from an
+        # overscan-subtracted overscan during ISR.
+        expectedKey = f"RESIDUAL STDEV {ampName}"
+
+        readNoises = []
+        for expMetadata in taskMetadata:
+            overscanNoise = expMetadata['isr'][expectedKey]
```

Comment:
I'll check for the key `isr` and if it doesn't exist, I'll continue the loop. If in the end the median readout noise can't be calculated, I'll return `None` (with a warining), and then in the function that estimates the gain, if the read noise is None but the correction type is not `NONE`, instead of raising (as it is right now), I'll issue a warning and default to correction type `None`, which does not require the noise, to at least return an estimation.